framework/src/systems/DisplacedSystem.C

```diff
@@ -46,22 +42,19 @@ DisplacedSystem::getVector(const std::string & name)
     return _undisplaced_system.getVector(name);
 }
 
-void
-DisplacedSystem::addTimeIntegrator(std::shared_ptr<TimeIntegrator> ti)
-{
-  _time_integrator = ti;
-}
-
-NumericVector<Number> &
-DisplacedSystem::solutionOldInternal() const
+const NumericVector<Number> &
+DisplacedSystem::getVector(const std::string & name) const
 {
-  return *_sys.old_local_solution;
+  if (_sys.have_vector(name))
+    return _sys.get_vector(name);
+  else
+    return _undisplaced_system.getVector(name);
```

Comment:
Should we error out here instead of returning the vector not for displaced system?

Reply:
The underlying interface in libmesh will error if not found.